pallets/automation-time/src/types.rs

```diff
@@ -323,4 +414,199 @@ mod tests {
+						execution_times: vec![0, 1].try_into().unwrap(),
+						executions_left: 2
+					}
+				);
+				assert_eq!(
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 0,
+						frequency: 3600
+					},
+					Schedule::Recurring { next_execution_time: 0, frequency: 3600 }
+				);
+				assert_ne!(
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 1,
+						frequency: 3600
+					},
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 0,
+						frequency: 3600
+					}
+				);
+				assert_ne!(
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 0,
+						frequency: 7200
+					},
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 0,
+						frequency: 3600
+					}
+				);
+				assert_ne!(
+					Schedule::<MaxExecutionTimes>::Fixed {
+						execution_times: vec![0].try_into().unwrap(),
+						executions_left: 1
+					},
+					Schedule::<MaxExecutionTimes>::Recurring {
+						next_execution_time: 0,
+						frequency: 3600
+					}
+				);
+			})
+		}
+
+		#[test]
+		fn new_fixed_schedule_sets_executions_left() {
+			new_test_ext(START_BLOCK_TIME).execute_with(|| {
+				let t1 = SCHEDULED_TIME + 3600;
+				let t2 = SCHEDULED_TIME + 3600 * 2;
+				let t3 = SCHEDULED_TIME + 3600 * 3;
+				let s = Schedule::<MaxExecutionTimes>::new_fixed_schedule::<Test>(vec![t1, t2, t3])
+					.unwrap();
+				if let Schedule::Fixed { executions_left, .. } = s {
+					assert_eq!(executions_left, 3);
+				} else {
+					panic!("Exepected Schedule::Fixed");
+				}
+			})
+		}
+
+		#[test]
+		fn new_fixed_schedule_errors_with_too_many_executions() {
+			new_test_ext(0).execute_with(|| {
+				let s = Schedule::<MaxExecutionTimes>::new_fixed_schedule::<Test>(
+					(0u64..=MaxExecutionTimes::get() as u64).collect(),
+				);
+				assert_err!(s, Error::<Test>::TooManyExecutionsTimes);
+			})
+		}
+
+		#[test]
+		fn new_fixed_schedule_cleans_execution_times() {
+			new_test_ext(START_BLOCK_TIME).execute_with(|| {
+				let t1 = SCHEDULED_TIME + 3600;
+				let t2 = SCHEDULED_TIME + 3600 * 2;
+				let t3 = SCHEDULED_TIME + 3600 * 3;
+				let s = Schedule::<MaxExecutionTimes>::new_fixed_schedule::<Test>(vec![
+					t1, t3, t2, t3, t3,
+				]);
+				if let Schedule::Fixed { execution_times, .. } = s.unwrap() {
+					assert_eq!(execution_times, vec![t1, t2, t3]);
+				} else {
+					panic!("Exepected Schedule::Fixed");
+				}
+			})
+		}
+
+		#[test]
+		fn checks_for_fixed_schedule_validity() {
+			new_test_ext(START_BLOCK_TIME).execute_with(|| {
+				assert_ok!(Schedule::<MaxExecutionTimes>::new_fixed_schedule::<Test>(vec![
+					SCHEDULED_TIME + 3600
+				]));
+				assert_err!(
+					Schedule::<MaxExecutionTimes>::new_fixed_schedule::<Test>(vec![
+						SCHEDULED_TIME + 3600,
+						SCHEDULED_TIME + 3650
+					]),
+					Error::<Test>::InvalidTime
+				);
+			})
+		}
+
+		#[test]
+		fn checks_for_recurring_schedule_validity() {
```

Comment:
I don't think you covered the case where the `frequency` is not a valid number. You where the frequency is 0, but not where it's say 3650.

Reply:
Good catch. Added a assertion for this.